prow/github/client.go

```diff
@@ -231,14 +231,16 @@ func (c *Client) Throttle(hourlyTokens, burst int) {
 //   An endpoint is used when all preceding endpoints have returned a conn err.
 //   This should be used when using the ghproxy GitHub proxy cache to allow
 //   this client to bypass the cache if it is temporarily unavailable.
-func NewClientWithFields(fields logrus.Fields, getToken func() []byte, bases ...string) *Client {
+func NewClientWithFields(fields logrus.Fields, getToken func() []byte, graphqlEndpoint string, bases ...string) *Client {
 	return &Client{
 		logger: logrus.WithFields(fields).WithField("client", "github"),
 		time:   &standardTime{},
-		gqlc: githubql.NewClient(&http.Client{
-			Timeout:   maxRequestTime,
-			Transport: &oauth2.Transport{Source: newReloadingTokenSource(getToken)},
-		}),
+		gqlc: githubql.NewEnterpriseClient(
```

Comment:
What are the side effects of doing this for non-enterprise customers? Should pick which to use more carefully?

Reply:
No side effect, this function wraps `NewClient()` and adds an endpoint url. Here is a link to this function: https://github.com/shurcooL/githubv4/blob/master/githubv4.go#L26

Reply:
I think the function is poorly named, but it is nice they didn't break signature.